drivers/sx127x/sx127x.c

```diff
@@ -339,6 +339,15 @@ static int _init_spi(sx127x_t *dev)
     /* Setup SPI for SX127X */
     res = spi_init_cs(dev->params.spi, dev->params.nss_pin);
 
+#ifdef MODULE_PERIPH_SPI_GPIO_MODE
+    spi_gpio_mode_t gpio_modes = {
+        .mosi = (GPIO_OUT | SX127X_DIO_PULL_MODE),
+        .miso = (SX127X_DIO_PULL_MODE),
+        .sclk = (GPIO_OUT | SX127X_DIO_PULL_MODE),
```

Comment:
For this to work you need to also do:    

```
    gpio_init_af(spi_config[bus].mosi_pin, spi_config[bus].af);
    gpio_init_af(spi_config[bus].miso_pin, spi_config[bus].af);
    gpio_init_af(spi_config[bus].sclk_pin, spi_config[bus].af);
```

Reply:
@aabadie Found the reason, when  calling `gpio_init(gpio_t pin, gpio_mode_t mode)` MODER register is set to mode (MODER = b'01), which erases the old Alternate function mode (MODER = b'10).